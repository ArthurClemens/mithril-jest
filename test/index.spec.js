/* global describe, it, expect */
import m from "mithril";
import { tidy } from "../index";

const page = {
  view: () =>
    m("div", [
      m("h1", "Page"),
      m("a", {
        href: "/home"
      }, "Back")
    ])
};

describe("Page component", () => {
  it("should have a title", () => {
    const cmp = m(page);
    const html = tidy(cmp);
    expect(html).toMatchSnapshot();
  });
  it("title should be h1 with label 'Page'", () => {
    const cmp = m(page);
    const html = tidy(cmp);
    expect(html).toContain("Page");
  });
});

