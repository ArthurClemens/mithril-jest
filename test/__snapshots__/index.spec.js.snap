exports[`Page component should have a title 1`] = `
"<div>
  <h1>
    Page
  </h1><a href=\"/home\">Back</a>
</div>
"
`;
